### PR TITLE
Try to get rid of e2e flake

### LINKED ISSUE #718: Expected 503 to be 200
```
STEP: creating the template (2018-11-12 12:50:12.309809182 +0000 UTC m=+9.448426721)
STEP: waiting for the template instance to turn ready (2018-11-12 12:50:13.065787269 +0000 UTC m=+10.204404817)
STEP: getting the route (2018-11-12 12:53:11.389313114 +0000 UTC m=+188.527930627)
STEP: hitting the route 3 times to check the counter (2018-11-12 12:53:11.418873873 +0000 UTC m=+188.557491462)

• Failure [187.884 seconds]
Openshift on Azure end user e2e tests [EndUser]
/go/src/github.com/openshift/openshift-azure/test/e2e/end_user_test.go:26
  should deploy a template and check the visit counter increments [It]
  /go/src/github.com/openshift/openshift-azure/test/e2e/end_user_test.go:62

  Expected
      <int>: 503
  to equal
      <int>: 200

  /go/src/github.com/openshift/openshift-azure/test/e2e/end_user_test.go:108
------------------------------
Openshift on Azure end user e2e tests [EndUser] 
  should disallow PDB mutations
  /go/src/github.com/openshift/openshift-azure/test/e2e/end_user_test.go:43
```
https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/openshift_openshift-azure/683/pull-ci-openshift-openshift-azure-master-e2e-azure/657
/kind test-flake

I think at least 500s should be retried

@jim-minter @thekad

### PR DESCRIPTION
**Pull request by thekad**
Fixes #718 
(hopefully)

* `wait.ForHTTPStatusOk` now returns the response too. The theory is that when the route is getting (re-) created there's a short period of time when one proxy isn't ready yet. By making the HTTP request retry until a 200 is returned, we side-step this issue
* couple of `go vet` changes

### CODE REVIEW

**Comment by thekad** [comment]
@mjudeikis yep, I'll wait until that one is merged

File: hack/loop.sh [hunk 1]
```diff
@@ -0,0 +1,41 @@
+#!/bin/bash
```
Comment (jim-minter) [inline hack/loop.sh]: I'd prefer this file wasn't checked in

File: pkg/util/wait/wait.go [hunk 2]
```diff
@@ -30,26 +32,27 @@ func PollImmediateUntil(interval time.Duration, condition wait.ConditionFunc, st
...
-// SimpleHttpClient to aid in mocking
-type SimpleHttpClient interface {
+// SimpleHTTPClient to aid in mocking
+type SimpleHTTPClient interface {
 	Do(req *http.Request) (*http.Response, error)
 }
 
 // ForHTTPStatusOk poll until URL returns 200
-func ForHTTPStatusOk(ctx context.Context, transport http.RoundTripper, urltocheck string) error {
+func ForHTTPStatusOk(ctx context.Context, transport http.RoundTripper, urltocheck string) ([]byte, error) {
```
Comment (jim-minter) [inline pkg/util/wait/wait.go]: return `(*http.Response, error)`

**Comment by mjudeikis** [comment]
/lgtm 
I will be surprised if this will fix the issue as I was under impression its bot the base but lets see :)

**Pull request merged**